plugins/services/src/js/structs/Framework.js

```diff
@@ -20,6 +20,14 @@ module.exports = class Framework extends Application {
     return this.get("slave_ids");
   }
 
+  getTasks() {
```

Comment:
@weblancaster @MatApple  Introducing an abstract getter like `getKey("tasks")` provides no guarantees and potentially introduces a high number of bugs thus should be avoided as discussed in on of our team syncs (reflection). 

I agree that we need to revisit our structs, but not because of the new (much more elegant) getter syntax or the seemingly redundant code.  
A struct is a composite type, which is used to model the relationship between types (e.g., the relationship between tasks and services as well as the relationship between service IDs and services). It provides the getters and setters to retrieve and assign values and should safeguard data access to prevent accidental mutations. We also use subtyping (`extend`) to describe that a struct is of the type of its parent struct. This allows us to add detail to a general data type and create a more specific data type (e.g. `Service` and `Framework`). These structs allow us to render all the different kinds of service in one list (service overview) as they share the same interface. 
Structs shouldn't transform data, considering the principles described,  which is the actual reason why we need to revisit our structs.